### PR TITLE
envd: Pass in self-hosted version, print in mz_version() (only)

### PR DESCRIPTION
**Pull request by def-**
See https://materializeinc.slack.com/archives/CQM08AP9V/p1729875799567639 for context
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/sql/src/session/vars.rs [hunk 1]
```diff
@@ -2441,7 +2441,7 @@ impl Var for BuildInfo {
     }
 
     fn value(&self) -> String {
-        self.human_version()
+        self.human_version(None)
```
Comment (benesch) [inline src/sql/src/session/vars.rs:2444]: This is the only one that seems a bit off to me—I think this is `SHOW mz_version`, right? And that seems like it ought to print the self-hosted version.

**Review by benesch — APPROVED** [review]
Seems about right to me! Not sure there's much we could do to improve this. Arguably we could fold this into `BuildInfo`, but it's a pain to do that because we need `&'static BuildInfo`s in a bunch of places...

File: src/sql/src/session/vars.rs (hunk 1 above)
Comment (def-) [reply src/sql/src/session/vars.rs:2444]: Apparently not:
```
materialize=> select mz_version();
                    mz_version
--------------------------------------------------
 v0.124.0-dev.0 (000000000, self-hosted: v25.1.0)
(1 row)
```

Comment (def-) [reply src/sql/src/session/vars.rs]:
```
materialize=> show mz_version;
                   mz_version
-------------------------------------------------
 v0.124.0-dev.0 (000000000, helm chart: v25.1.0)
(1 row)

materialize=> select mz_version();
                   mz_version
-------------------------------------------------
 v0.124.0-dev.0 (000000000, helm chart: v25.1.0)
(1 row)
```
Fixed

**Review by benesch — APPROVED** [review]
Looks great—thanks, @def-!

**Pull request merged**